### PR TITLE
WT-10231 Extra RTS assertions

### PR DESCRIPTION
**Pull request by korteland**
I considered converting a bunch of `WT_ASSERT`s into `WT_ASSERT_ALWAYS`, but to be honest I don't know enough to be confident about pushing those assertions into the real world.

### CODE REVIEW

File: src/rollback_to_stable/rts_api.c [hunk 1]
```diff
@@ -8,6 +8,24 @@
 
 #include "wt_internal.h"
 
+/*
+ * __rts_assert_timestamps_unchanged --
+ *     Wrapper for some diagnostic assertions related to timestamps.
```
Comment (kommiharibabu) [inline src/rollback_to_stable/rts_api.c:13]: how about changing it from `timestamps` to `global timestamps`?

**Review by kommiharibabu — APPROVED** [review]
LGTM

File: src/rollback_to_stable/rts_api.c (hunk 1 above)
Comment (korteland) [reply src/rollback_to_stable/rts_api.c]: Yep fair enough. Done in 8f205f6.

**Comment by korteland** [comment]
evergreen merge

**Comment by korteland** [comment]
Not sure what happened with the merge commit - I can't find that issue in Jira, but it didn't leave a core dump (or print a backtrace) so I don't know what the issue was. I spent a bunch of time testing it locally and it seems OK. I'll merge this, since we can always revert if that failure turns out to be a real issue.

**Pull request merged**